packages/blade/src/components/Box/BaseBox/getBaseBoxStyles.ts

```diff
@@ -1,102 +1,239 @@
 import type { CSSObject } from 'styled-components';
-import type { BaseBoxProps } from './types';
-import { getIn, makeBorderSize, makeSize, makeSpace } from '~utils';
+import type {
+  BaseBoxProps,
+  MakeValueResponsive,
+  SpacingValueType,
+  ArrayOfMaxLength4,
+} from './types';
+import type { Breakpoints } from '~tokens/global/breakpoints';
+import breakpoints from '~tokens/global/breakpoints';
+import { getMediaQuery } from '~src/utils/getMediaQuery';
+import { isReactNative, isEmpty, getIn, makeSpace, makeSize } from '~utils';
 import type { Theme } from '~components/BladeProvider';
 
-// allows 'auto' as spacing value
-const getSpacingValue = <SpacingValue extends string | number | undefined>({
-  value,
-  theme,
-}: {
-  value: SpacingValue;
-  theme: Theme;
-}): string | undefined => {
-  if (value === 'auto') return 'auto';
-  /**
-   * NOTE: don't allow numbers once we have the scale for sizing
-   * DO NOT PUBLISH THIS COMPONENT PUBLICLY WITH THIS PIECE OF CODE
-   */
-  if (typeof value === 'number') {
-    return makeSpace(value);
+const getResponsiveValue = <T extends string | number | string[]>(
+  value: MakeValueResponsive<T> | undefined,
+  size: keyof Breakpoints = 'base',
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+): any | undefined => {
+  if (value === undefined || value === null) {
+    return undefined;
   }
-  return value ? makeSpace(getIn(theme, value)) : undefined;
+
+  if (typeof value === 'string' || typeof value === 'number' || Array.isArray(value)) {
+    if (size === 'base') {
+      return value;
+    }
+
+    // Plain values will already get added in base styles so we don't have to repeat them in media queries
+    return undefined;
+  }
+
+  if (isEmpty(value)) {
+    return undefined;
+  }
+
+  if (isReactNative()) {
+    // In React Native, we map the value `s` token on priority (since the breakpoint maps to mobiles in useBreakpoint hook).
+    // We further look into smaller sizes, then we check base size.
+    // Then we return the first non-undefined value in this priority
```

Comment:
- Ok so react native doesn't have media queries. 
- So if someone does this - `<Box padding={{  s: 'spacing.0', m: 'spacing.10' }} />` How do we handle it?
   Ans: We pick `s` because that is the first matching point for mobile
- What about `<Box padding={{ base: 'spacing.1', xs: 'spacing.0', m: 'spacing.10' }} />`?
  Ans: We pick `xs` in above example because that is the first matching mobile breakpoint now.
- What about `<Box padding={{ base: 'spacing.1', m: 'spacing.10' }} />`?
  Ans: We pick `base` here because we have not defined any mobile matching point so we default to `base`
- What about `<Box padding={{ m: 'spacing.10' }} />`?
  Ans: We should set `undefined` because we have not defined any value. The value defined on `m` is for desktop styles so it should be ignored on mobile because on web it will be ignored.


So to solve this, we come with this priority logic
- We look for breakpoints in this sequence -> `s > xs > base` and return the first undefined value. Else we just return undefined.

This will mean when you're writing `<Box padding={{ base: 'spacing.0', s: 'spacing.1', m: 'spacing.10' }} />` it works the same way on web and native both.

Now this logic is majorly for us internally and for consumers that write cross-platform code. People who only work on react native, wouldn't care about responsiveness and will define the plain value properties.

Reply:
Can we document this in both storybook as well as somewhere in this function?